src/scripts/main.js

```diff
@@ -1,3 +1,92 @@
 'use strict';
 
-// write code here
+const buttonContainer = document.querySelector('.container');
+const maxRow = 10;
+const minRow = 2;
+const maxColumn = 10;
+const minColumn = 2;
+
+buttonContainer.addEventListener('click', (e) => {
+  const target = e.target;
+  const tbody = document.querySelector('.field tbody');
+  const className = target.className.slice(0, -7);
+
+  switch (className) {
+    case 'append-row':
+      if (tbody.children.length < maxRow) {
+        const newRow = tbody.lastElementChild.cloneNode(true);
+
+        tbody.append(newRow);
+
+        if (tbody.children.length >= maxRow) {
+          target.setAttribute('disabled', true);
+        }
+
+        if (tbody.children.length > minRow) {
+          const removeRowButton = buttonContainer.querySelector('.remove-row');
```

Comment:
I suppose you can take it outside of the function and not assign the variable each time. You should know that each DOM operations include `querySelector` are heavy.

Reply:
При выполнении скрипта мы попадем лишь в один какой-нибудь кейс из четырех и, как следствие, будем искать в дом-дереве только один элемент. В варианте, который предлагаешь ты, будут искаться сразу все четыре элемента. Зачем это делать? В чем логика?

Reply:
in case, that I suggested, we will define all this variables 1 time when app run. in case you implemented, if we click on button 10 times, it will query node 10 times, but we can find it only once

Reply:
Fixed creation of variables